dashboard/tests/test_dashboard.py

```diff
@@ -156,6 +157,24 @@ def _search_agent(processes):
     assert agent_ports is not None
 
 
+@pytest.mark.parametrize(
+    "ray_start_with_dashboard", [{
+        "dashboard_host": "127.0.0.1"
+    }, {
+        "dashboard_host": "0.0.0.0"
+    }, {
+        "dashboard_host": "::"
+    }],
+    indirect=True)
+def test_dashboard_address(ray_start_with_dashboard):
+    webui_url = ray_start_with_dashboard["webui_url"]
+    webui_ip = webui_url.split(":")[0]
+    assert not ipaddress.ip_address(webui_ip).is_unspecified
+    assert webui_ip in [
+        "127.0.0.1", ray_start_with_dashboard["node_ip_address"]
+    ]
+
```

Comment:
We should also verify that the ip:port stored in redis is correct.

Reply:
The webui_url returned from `ray.init()` is fetched from redis. Please refer to: https://github.com/ray-project/ray/blob/master/python/ray/_private/services.py#L1221